CMakeScripts/JoinPaths.cmake

```diff
@@ -12,10 +12,12 @@ function(join_paths joined_path first_path_segment)
     set(temp_path "${first_path_segment}")
     foreach(current_segment IN LISTS ARGN)
         if(NOT ("${current_segment}" STREQUAL ""))
-            if(IS_ABSOLUTE "${current_segment}")
-                set(temp_path "${current_segment}")
+            string(REGEX REPLACE "^${CMAKE_INSTALL_PREFIX}" "" tmp_segment "${current_segment}")
+            if(NOT IS_ABSOLUTE "${current_segment}" OR NOT "${current_segment}" MATCHES "^${tmp_segment}$")
```

Comment:
It is a little hard to reason about, swapping the order. It would be nicer to store the `NOT "${current_segment}" MATCHES "^${tmp_segment}$"` to a separate boolean variable named `starts_with_cmake_install_prefix`. Or at least rename `tmp_segment` to `current_segment_with_install_prefix_stripped`.

Reply:
I got rid of `tmp_segment` entirely in favor of `new_segment`. I can change the variable name still if you want?

Maybe its my lack of experience with cmake, but storing `NOT "${current_segment}" MATCHES "^${tmp_segment}$"` in a bool variable seems to needlessly make the script either more complicated or verbose. I also seems to be the most simple solution to me if `IS_ABSOLUTE` is the `else` case, but please help me understand if I'm not getting it. :)